nanomi-optics/nanomi_optics/gui/window_main.py

```diff
@@ -175,3 +192,47 @@ def u_lens_mode(self):
         elif md == "Cf":
             self.mode = True
         self.upper_menu.set_mode(self.mode)
+
+    def save_results(self):
+        """save results in csv file"""
+        file_path = tk.filedialog.askdirectory()
+        file_path = file_path + "/nanomi_results.csv"
```

Comment:
Never join paths like this. It has undefined behaviour on different operating systems. `os.path.join` is the correct way.